doc/dev/debugging.rst

```diff
@@ -403,4 +403,42 @@ The recompilation starts after the given checkpoint stage.
 
     replace_ir(f, "HLOLoweringPass", new_ir)
     f(2.0)
-    >> 8.0
\ No newline at end of file
+
+C Executable Generation
+=================
+
+Catalyst provides a way to generate a c executable that calls the shared object of the given
+qjit-decorated function.
+In ``catalyst.debug.compiler_functions``, ``compile_executable(fn: QJIT, *args) -> str, str`` takes
+a qjit-decorated function and arguments for that function.
+It returns a string of LD_LIBRARY_PATH and the path to the output executable file.
+Users can use `debug.print_memref` to add information to stdout.
+
+The following example is a square function.
+Here we are using debug.print_memref to print the information of the result from ``y``.
+Note that ``keep_intermediate=True`` help us to keep the generated files so that we
+can re-run the executable latter.
+
+.. code-block:: python
+    @qjit(keep_intermediate=True)
```

Comment:
It would be nice to generate the executable in the current working directory if `keep_intermediate=False` as a slight improvement in usability, is that possible?

Reply:
Yes. The compiler has already been able to generate executable with `keep_intermediate=False`. The file will appear in the project root. I have added some sentences for that in docs.